### PR TITLE
Add stubs for DataDeviceManager & DataDevice.

### PR DESCRIPTION
**Pull request by AlanGriffiths**
These will be needed for xdg-shell support.

### CODE REVIEW

File: src/server/frontend/wayland/wayland_connector.cpp [hunk 1]
```diff
@@ -2398,6 +2442,7 @@ mf::WaylandConnector::WaylandConnector(
         display.get(),
         display_config);
     shell_global = std::make_unique<mf::WlShell>(display.get(), shell, *seat_global);
+    data_device_manager_global = std::make_unique<DataDeviceManager>(display.get(), wl_data_device_interface.version);
```
Comment (RAOF) [inline src/server/frontend/wayland/wayland_connector.cpp]: I don't think this is safe - if the wl_data_device protocol ever gets a new version, the version we claim to support will change, without us actually supporting the new version.

At best the new version has a new method, and clients which call that method crash the server. At worst, there's a purely semantic change which results in some clients mysteriously failing to work properly. 😁

We need to always explicitly write our max_version; any change to that number must require code changes anyway.

File: src/server/frontend/wayland/wayland_connector.cpp [hunk 2]
```diff
@@ -2186,6 +2186,50 @@ class WlShell : public wayland::Shell
...
+    }
+
+    void release() override
+    {
+    }
+};
+
+struct DataDeviceManager : wayland::DataDeviceManager
+{
+    DataDeviceManager(struct wl_display* display, uint32_t max_version) :
```
Comment (RAOF) [inline src/server/frontend/wayland/wayland_connector.cpp]: I don't think that max_version should be an externally-visible constructor argument; the code in DataDeviceManager either supports a given protocol version, or it doesn't, and we always have to support clients binding to a lower version than we support, too.

I think we should strive for the protocol-implementing objects to contain the versioning differences as much as possible.

**Comment by RAOF** [comment]
Other than the inline comments, looks good.

**Comment by AlanGriffiths** [comment]
bors try

**Comment by Saviq** [comment]
bors try

**Pull request merged**